libsolidity/interface/CompilerStack.cpp

```diff
@@ -70,6 +72,33 @@ void CompilerStack::setRemappings(vector<string> const& _remappings)
 		r.prefix = colon == eq ? string(remapping.begin(), eq) : string(colon + 1, eq);
 		r.target = string(eq + 1, remapping.end());
 		remappings.push_back(r);
+		if (r.context.empty())
+		{
+			string prefix = r.prefix;
+			index.emplace(prefix, remappings.size()-1);
+		}
+		else
+		{
+			string context = r.context;
+			contextReferences.emplace(context, remappings.size()-1);
+		}
+	}
+	// fill references
+	for (auto const& key: contextReferences)
+	{
+		string reference = key.first;
+		if (reference.front() != '@')
+			continue; // ignore
+		reference = string(reference.begin()+1, reference.end());
+		try
+		{
+			string referencedTarget = remappings.at(index.at(reference)).target;
+			remappings[key.second].context = referencedTarget;
+		}
+		catch (...)
+		{
+			continue;
```

Comment:
Isn't this a point to report an error `No prefix "non_existent" is globally visible, but @non_existent is referenced.`